tools/engine_tool/lib/src/build_utils.dart

```diff
@@ -65,6 +65,31 @@ void debugCheckBuilds(List<Build> builds) {
   }
 }
 
+/// Transform the name of a build into the name presented and accepted by the
+/// CLI
+///
+/// If a name starts with '$OS/', it is a local development build, and the
+/// mangled name has the '$OS/' part stripped off.
+///
+/// If the name does not start with '$OS/', then it starts with 'ci/' and is
+/// returned unchanged.
+String mangleConfigName(Environment environment, String name) {
+  final String os = '${environment.platform.operatingSystem}/';
+  return name.startsWith(os) ? name.substring(os.length) : name;
+}
+
+/// Transform the mangled (CLI) name of a build into its true name in the build
+/// config json file.
+///
+/// This does the reverse of [mangleConfigName] taking the operating system
+/// name from `environment`.
+String demangleConfigName(Environment environment, String name) {
+  const String cif = 'ci/';
+  const String cib = r'ci\';
```

Comment:
Is this for Windows or something else, I don't understand this one :P

Reply:
Yes, I would rename cif and cib to something that indicates Win/Unix path endings.

Also you could factor these constants out and use them above on line 82.

Reply:
If so I think we could use `package:path/path.dart#fragment` (or just rely on the methods in `path` that implicitly use the right fragment).

Reply:
Cleaned this up using these suggestions. Thanks!